### PR TITLE
added signal_to_noise_threshold option to remove_noise function

### PR DESCRIPTION
**Pull request by StSchulze**
- allows to define a signal_to_noise_threshold when using the remove_noise function
- default behavior is the same as before, i.e. the noise level can be defined either directly (key word noise_level) or by estimating the noise level after defining a mode for this, and default S/N threshold is 1.0
- merged handling of specs without binary tags

### CODE REVIEW

File: tests/main_spec_test.py [hunk 1]
```diff
@@ -384,6 +384,10 @@ def test_peaks(self):
             ],
         )
 
+    def test_estimate_noise_mean_spec_6(self):
+        spec = self.Run[6]
+        self.assertEqual(90407.70511570283, spec.estimated_noise_level("mean"))
```
Comment (MKoesters) [inline tests/main_spec_test.py]: I would rather use `self.assertAlmostEqual` when comparing floats

File: pymzml/spec.py [hunk 2]
```diff
@@ -1282,13 +1285,13 @@ def remove_noise(self, mode="median", noise_level=None):
...
         if noise_level is None:
             noise_level = self.estimated_noise_level(mode=mode)
-        if self._peak_dict["centroided"] is not None:
+        if len(self._peak_dict["centroided"]) != 0:
             self._peak_dict["centroided"] = self.peaks("centroided")[
-                self.peaks("centroided")[:, 1] >= noise_level
+                self.peaks("centroided")[:, 1]/noise_level >= signal_to_noise_threshold
             ]
-        if self._peak_dict["raw"] is not None:
+        if len(self._peak_dict["raw"]) != 0:
```
Comment (MKoesters) [inline pymzml/spec.py]: As discussed, I think `len(self.peaks('centroided')) != 0` would be better

File: tests/main_spec_test.py (hunk 1 above)
Comment (StSchulze) [reply tests/main_spec_test.py]: Done

File: pymzml/spec.py (hunk 2 above)
Comment (StSchulze) [reply pymzml/spec.py]: Done

**Pull request merged**